packages/core/src/util/firmware.ts

```diff
@@ -9,6 +9,28 @@ export interface Firmware {
 	firmwareTarget?: number;
 }
 
+/**
+ * Guess the firmware format based on filename and firmware buffer
+ *
+ * @param filename The firmware filename
+ * @param firmware The firmware file data buffer
+ */
+export function guessFirmwareFormat(
+	filename: string,
+	firmware: Buffer,
+): FirmwareFileFormat {
+	if (
+		(filename.endsWith(".exe") || filename.endsWith(".ex_")) &&
+		firmware.includes(Buffer.from("Aeon Labs", "utf8"))
+	) {
+		return "aeotec";
+	} else if (/\.(hex|ota|otz)$/.test(filename)) {
+		return filename.slice(-3) as FirmwareFileFormat;
+	} else {
+		throw new Error("could not detect firmware format");
```

Comment:
This should be a `ZWaveError` with a corresponding `ZWaveErrorCode`.

Reply:
fixed